apps/arena/lib/arena/matchmaking/pair_mode.ex

```diff
@@ -53,21 +53,58 @@ defmodule Arena.Matchmaking.PairMode do
   @impl true
   def handle_info(:launch_game?, %{clients: clients} = state) do
     Process.send_after(self(), :launch_game?, 300)
+
+    state =
+      if Map.has_key?(state, :game_mode_configuration) do
+        state
+      else
+        case Arena.Configuration.get_game_mode_configuration("pair", "battle_royale") do
+          {:error, _} ->
+            state
+
+          {:ok, game_mode_configuration} ->
+            # This is needed because we might not want to send a request every 300 seconds to the game backend
+            map = Enum.random(game_mode_configuration.map_mode_params)
+
+            Process.send_after(self(), :update_params, 5000)
+
+            Map.put(state, :game_mode_configuration, game_mode_configuration)
+            |> Map.put(:current_map, map)
+        end
+      end
```

Comment:
We could create a function for this block of code that we use in every matchmaking.

Reply:
done in [refactor: modularize function in matchmaking module](https://github.com/lambdaclass/mirra_backend/pull/1061/commits/39f35cae42fb70e3b8acb37627e6c15a9a97ae1c)